File: GoodWill_Spring_BackEnd/src/main/java/com/GoodWillCharity_SprinBoot_API/app/service/IAdminService.java
package com.GoodWillCharity_SprinBoot_API.app.service;



import java.util.concurrent.CompletableFuture;

import com.GoodWillCharity_SprinBoot_API.app.model.Admin;

public interface IAdminService{
    CompletableFuture<Void> createAsync(Admin admin);

    CompletableFuture<Admin> loginAdmin(String email, String password);

    String generateToken(Admin admin);
}

File: GoodWill_DotNet_BackEnd/GoodWillCharity_DotNet_API/Services/Abstract/IAdminService.cs
using GoodWillCharity_DotNet_API.Models;

namespace GoodWillCharity_DotNet_API.Services.Abstract
{
    public interface IAdminService
    {
        Task CreateAsync(Admin admin);
        Task<Admin> LoginAdmin(string email, string password);
        string GenrateToken(Admin admin);

    }
}
